server/src/main/java/org/elasticsearch/common/geo/BoundingBox.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.geo;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.geometry.Geometry;
+import org.elasticsearch.geometry.Rectangle;
+import org.elasticsearch.geometry.ShapeType;
+import org.elasticsearch.geometry.utils.StandardValidator;
+import org.elasticsearch.geometry.utils.WellKnownText;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Objects;
+
+/**
+ * A class representing a Geo-Bounding-Box for use by Geo queries and aggregations
+ * that deal with extents/rectangles representing rectangular areas of interest.
+ */
+public class BoundingBox implements ToXContentObject, Writeable {
```

Comment:
Since this is only for Geographic logic (e.g., dateline crossing) we should probably rename to `GeoBoundingBox`? 
```suggestion
public class GeoBoundingBox implements ToXContentObject, Writeable {
```

Reply:
The main reason I resisted this change is because I thought it was too close to the `GeoBoundingBox` query. since there is also even a `GeoBoundingBoxIT`.

Reply:
I've renamed `GeoBoundingBoxIT` to `GeoBoundingBoxQueryIT` to help make things more explicit